rush-plugins/rush-azure-storage-build-cache-plugin/src/KeyVaultAuthentication.ts

```diff
@@ -0,0 +1,92 @@
+import type { DeviceCodeCredential } from '@azure/identity';
+import { type KeyVaultSecret, SecretClient } from '@azure/keyvault-secrets';
+import type { ITerminal } from '@rushstack/node-core-library';
+import type { ICredentialResult, IAzureAuthenticationBaseOptions } from './AzureAuthenticationBase';
+// Importing this from the index triggers a runtime import of `@rushstack/rush-sdk`.
+import { AzureAuthenticationBase } from './AzureAuthenticationBase';
+import type { ICredentialCacheEntry } from '@rushstack/rush-sdk';
+
+/**
+ * @public
+ */
+export interface IKeyVaultAuthenticationOptions extends IAzureAuthenticationBaseOptions {
+  updateCredentialCommand?: string;
+  vaultName: string;
+  secretName: string;
+  /**
+   * If this value is sooner than the expiration from KeyVault, use this time instead.
+   *
+   * @remarks
+   * If the expiration from KeyVault is sooner, that time is used instead.
+   */
+  overrideExpiration?: Date;
+}
+
+interface IKeyVaultCredentialMetadata {
+  credentialVersion: string | undefined;
+}
+
+/**
+ * @public
+ */
+export class KeyVaultAuthentication extends AzureAuthenticationBase {
```

Comment:
This probably should go in whatever code would make use of this instead of in this package.